File: packages/rest-api/src/models/flight.rs
use rocket::serde::Serialize;
use rocket_db_pools::sqlx;
use sqlx::types::chrono;
use sqlx::FromRow;

#[derive(Serialize, FromRow)]
#[sqlx(rename_all = "camelCase")]
#[serde(crate = "rocket::serde")]
pub struct Flight {
    pub id: sqlx::types::Uuid,
    pub callsign: String,
    pub beginDate: sqlx::types::chrono::DateTime<chrono::Utc>,
    pub endDate: Option<sqlx::types::chrono::DateTime<chrono::Utc>>,
    pub company: String,
    pub flightNumber: i32,
    pub aircraft: serde_json::Value,
    pub departureIcao: String,
    pub estimatedOffBlockTime: String,
    pub cruisingSpeed: String,
    pub remarks: String,
    pub flightRules: String,
    pub estimatedEnrouteMinutes: i32,
    pub arrivalIcao: String,
    pub route: String,
    pub cruisingLevel: i32,
    pub weekdays: serde_json::Value,
    pub updatedAt: sqlx::types::chrono::DateTime<chrono::Utc>,
    pub createdAt: sqlx::types::chrono::DateTime<chrono::Utc>,
}

File: packages/common/src/flight.ts
import { FlightRules, WakeTurbulence, Weekdays } from "./enum";

export * from "./enum";

type Flight = {
  id: string;
  callsign: string;
  beginDate: Date;
  endDate: Date | null;
  company: string;
  flightNumber: number;
  aircraft: {
    icaoCode: string;
    equipment: string;
    wakeTurbulence: WakeTurbulence;
  };
  departureIcao: string;
  estimatedOffBlockTime: string;
  cruisingSpeed: string;
  weekdays: Weekdays[];
  cruisingLevel: number;
  route: string;
  arrivalIcao: string;
  estimatedEnrouteMinutes: number;
  remarks: string;
  flightRules: FlightRules;
};

export default Flight;
